code.js

```diff
@@ -1,3 +1,14 @@
-function fib(n) {
-    return n;
+function fib(n, fib_solns = {}) {
+    if (n == 0) return [0];
+    if (n == 1) return [0, 1]; //base cases <-^
+
+    if (!fib_solns[n]) { //if n is not in the array
+        const leftNum1 = fib(n - 1, fib_solns); //recursively compute n - 1
```

Comment:
Could you leverage the numbers already in the array here instead of making recursive calls?